modules/database/src/ioc/db/dbDbLink.c

```diff
@@ -179,49 +175,19 @@ static long dbDbGetValue(struct link *plink, short dbrType, void *pbuffer,
             return status;
     }
 
-    if (ppv_link->getCvt && ppv_link->lastGetdbrType == dbrType)
-    {
-        /* shortcut: scalar with known conversion, no filter */
-        status = ppv_link->getCvt(dbChannelField(chan), pbuffer, paddr);
-    }
-    else if (dbChannelFinalElements(chan) == 1 && (!pnRequest || *pnRequest == 1)
-                && dbChannelSpecial(chan) != SPC_DBADDR
-                && dbChannelSpecial(chan) != SPC_ATTRIBUTE
-                && ellCount(&chan->filters) == 0)
-    {
-        /* simple scalar: set up shortcut */
-        unsigned short dbfType = dbChannelFinalFieldType(chan);
-
-        if (dbrType < 0 || dbrType > DBR_ENUM || dbfType > DBF_DEVICE)
-            return S_db_badDbrtype;
-
-        ppv_link->getCvt = dbFastGetConvertRoutine[dbfType][dbrType];
-        ppv_link->lastGetdbrType = dbrType;
-        status = ppv_link->getCvt(dbChannelField(chan), pbuffer, paddr);
-    }
-    else
-    {
-        /* filter, array, or special */
-        ppv_link->getCvt = NULL;
-
-        if (ellCount(&chan->filters)) {
-            /* If filters are involved in a read, create field log and run filters */
-            pfl = db_create_read_log(chan);
-            if (!pfl)
-                return S_db_noMemory;
-
-            pfl = dbChannelRunPreChain(chan, pfl);
-            pfl = dbChannelRunPostChain(chan, pfl);
-        }
-
-        status = dbChannelGet(chan, dbrType, pbuffer, NULL, pnRequest, pfl);
-
-        if (pfl)
-            db_delete_field_log(pfl);
-
-        if (status)
-            return status;
+    /* If filters are involved in a read, create field log and run filters */
+    if (ellCount(&chan->filters)) {
+        pfl = db_create_read_log(chan);
```

Comment:
Maybe I've misunderstood the [discussion](https://code.launchpad.net/~dirk.zimoch/epics-base/+git/epics-base/+merge/378968), but it sounds like `db_create_read_log()` is what really needs to be optimized.  (which makes more sense to me wrt. global locking)  Maybe by adding a way to allow the first (and often only) `db_field_log` to be created on the stack?  eg. by adding a bit to control whether `db_delete_field_log()` calls `freeListFree()`.

Reply:
Ok, the immediate issue with what I suggest would be filters like `dbnd` which store `db_field_log*`.